### DISCUSSION TITLE
How to encode a field using conditions in Vega-Lite

### DISCUSSION THREAD

**Discussion by nickjwheatley** [vega/vega-lite#8673] [Q&A] [answered]
OBJECTIVE

Hi Folks, I'm trying to add data labels to my chart, however I have multiple bars layered on top of each other, and I need the data labels to hover over different bars depending on if values in a certain field are positive or negative.

ATTEMPT

This could be achieved by changing the "field" property of the "y" encoding using a condition. I've spent some time exploring [Vega-Lite documentation](https://vega.github.io/vega-lite-v2/docs/condition.html) and experimenting with some stuff, but I can't get it to work no matter what I try. Vega seems to ignore by condition. I'm also curious if I'm able to apply conditions to "mark" rather than "encoding". When values are negative, I'd like to change "dy" to 10 from -10.

Any suggestions?

```
 "mark": {
    "type":"text",
    "dy":-10
  },
  "encoding": {
    "text": {
      "field": "field_one"
    },
    "y": {
      "condition":{
        "test":"datum['test_data'] < 0",
        "field": "field_one"
      }, 
      "field": "field_two"
    }....
  }
```
![image](https://user-images.githubusercontent.com/55959180/214755159-67d738c5-1b2e-4b99-9e81-723b271a8da5.png)

**Comment by chanwutk** [comment, marked as answer]
For your `dy` question, would you mind checking this [example](https://vega.github.io/vega-lite/examples/bar_negative_horizontal_label.html)?

For the other question, would you mind trying to transform with `calculate` before encoding?

Here is an example
```json
  "mark": {
    "type":"text",
    "dy": {"expr": "datum.longer_bar_value < 0 ? 10 : -10"}
  },
  "transform": {
    {
      "calculate": "Math.abs(datum.thin_bar_value) > Math.abs(datum.thick_bar_value) ? datum.thin_bar_value : datum.thick_bar_value",
      "as": "longer_bar_value"
    }
  }
  "encoding": {
    "text": {
      "field": "longer_bar_value"
    },
    "y": {
      "field": "longer_bar_value",
      "type": "quantitative"
    }....
  }
```

Please let us know if you have any question :)
